vllm/transformers_utils/config.py

```diff
@@ -232,20 +232,76 @@ def get_config(
     return config
 
 
+def get_hf_file_to_dict(file_name, model, revision):
+    """
+    Downloads a file from the Hugging Face Hub and returns 
+    its contents as a dictionary.
+
+    Parameters:
+    - file_name (str): The name of the file to download.
+    - model (str): The name of the model on the Hugging Face Hub.
+    - revision (str): The specific version of the model. 
+
+    Returns:
+    - config_dict (dict): A dictionary containing 
+    the contents of the downloaded file.
+    """
+    file_path = Path(model) / file_name
+
+    if not file_path.is_file():
+        file_path = Path(hf_hub_download(model, file_name, revision=revision))
+
+    with open(file_path, "r") as file:
+        config_dict = json.load(file)
+
+    return config_dict
+
+
+def get_pooling_config(model, revision='main'):
+    """
+    This function gets the pooling and normalize 
+    config from the model.
```

Comment:
Can we rename this so it's more obvious that this only applies to sentence-transformers models? We currently already have embedding models that are not from sentence-transformers, so it's rather confusing why only BERT has `PoolingConfig`.

Reply:
Sorry about the force push!! I made a rebase with main to solve the conflicts.